### PR TITLE
stm32f7: add SocketCAN support

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- stm32f7/nucleo-144: add CAN1 pins selection
- stm32f7/nucleo-144: refactor bringup logic
- stm32f7: add SocketCAN support
- nucleo-144/f722: add CAN and SocketCAN configurations

## Testing
nucleo-144/f722-cansock with canutils/cansend and canutils/candump tools

### CODE REVIEW

File: boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+include $(TOPDIR)/.config
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs:21]: why not share the same Make.defs for all configs

File: boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+############################################################################
+# boards/arm/stm32f7/nucleo-144/configs/f746-nsh/Make.defs
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs:2]: wrong path

File: boards/arm/stm32f7/nucleo-144/configs/f722-cansock/Make.defs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+############################################################################
+# boards/arm/stm32f7/nucleo-144/configs/f746-nsh/Make.defs
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/configs/f722-cansock/Make.defs:2]: ditto

File: arch/arm/src/stm32f7/stm32_can_sock.c [hunk 4]
```diff
@@ -0,0 +1,2538 @@
...
+   *
+   * - Disables silent mode.
+   * - Disables loopback mode.
+   *
+   * NOTE that for the time being, SJW is set to 1 just because I don't
+   * know any better.
+   */
+
+  tmp = ((brp - 1) << CAN_BTR_BRP_SHIFT) | ((ts1 - 1) << CAN_BTR_TS1_SHIFT) |
+        ((ts2 - 1) << CAN_BTR_TS2_SHIFT) | ((1 - 1) << CAN_BTR_SJW_SHIFT);
```
Comment (pkarashchenko) [inline arch/arm/src/stm32f7/stm32_can_sock.c:1887]: why (1 - 1) is needed? maybe better to add `const uint32_t sjw = 1;` just for readability?

Comment (raiden00pl) [reply arch/arm/src/stm32f7/stm32_can_sock.c:1887]: I prefer to keep it as it is for compliance with other STM32 CAN implementations

File: boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs (hunk 1 above)
Comment (raiden00pl) [reply boards/arm/stm32f7/nucleo-144/configs/f722-can/Make.defs:21]: All configurations in stm32f7/nucleo-144 look like this. 
Personally, I think stm32f7/nucleo-144 should be divided into separate boards as it is done for the rest of Nucleo boards.
The current solution is misleading - there are nucleo-144 boards that are not based on stm32f7.  
I don't know if any other boards family has a similar organization.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**